src/data/equiv/list.lean

```diff
@@ -117,6 +117,32 @@ def fintype_pi (α : Type*) (π : α → Type*) [fintype α] [decidable_eq α] [
   (@fintype_arrow α (Σa, π a) _ _ (@encodable.sigma _ _ a _)).bind $ λf,
   trunc.mk $ @encodable.of_equiv _ _ (@encodable.subtype _ _ f _) (equiv.pi_equiv_subtype_sigma α π)
 
+/-- The elements of a `fintype` as a sorted list. -/
+def sorted_univ (α) [fintype α] [encodable α] : list α :=
+finset.univ.sort (encodable.encode' α ⁻¹'o (≤))
+
+theorem mem_sorted_univ {α} [fintype α] [encodable α] (x : α) : x ∈ sorted_univ α :=
+(finset.mem_sort _).2 (finset.mem_univ _)
+
+theorem length_sorted_univ {α} [fintype α] [encodable α] : (sorted_univ α).length = fintype.card α :=
+finset.length_sort _
+
+theorem sorted_univ_nodup {α} [fintype α] [encodable α] : (sorted_univ α).nodup :=
+finset.sort_nodup _ _
+
+/-- An encodable `fintype` is equivalent a `fin`.-/
+def fintype_equiv_fin {α} [fintype α] [encodable α] :
+  α ≃ fin (fintype.card α) :=
+begin
+  haveI : decidable_eq α := encodable.decidable_eq_of_encodable _,
+  rw ← @length_sorted_univ α,
```

Comment:
I didn't check this in an editor, but this is using a `rw` to define data, right? Maybe better to avoid it and use `equiv.cast` instead?